packages/compass-settings/src/stores/settings.ts

```diff
@@ -1,50 +1,49 @@
 import type { Reducer } from 'redux';
-import type { RootState } from '.';
-import type { ThunkAction, ThunkDispatch } from 'redux-thunk';
+import type { SettingsThunkAction } from '.';
 import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
 import type {
   PreferenceStateInformation,
   UserConfigurablePreferences,
-  PreferencesAccess,
 } from 'compass-preferences-model';
-import preferences from 'compass-preferences-model';
-import { pick } from '../utils/pick';
 
 const { log, mongoLogId } = createLoggerAndTelemetry('COMPASS-SETTINGS');
 
-export type State =
+export type State = { isModalOpen: boolean } & (
   | {
       loadingState: 'loading';
-      sandbox: null;
       settings: Record<string, never>;
       preferenceStates: Record<string, never>;
       updatedFields: never[];
     }
   | {
       loadingState: 'ready';
-      sandbox: PreferencesAccess;
       settings: UserConfigurablePreferences;
       preferenceStates: PreferenceStateInformation;
       updatedFields: (keyof UserConfigurablePreferences)[];
-    };
+    }
+);
 
 export const INITIAL_STATE: State = {
-  sandbox: null,
+  isModalOpen: false,
   settings: {},
   preferenceStates: {},
   updatedFields: [],
   loadingState: 'loading',
 };
 
 export enum ActionTypes {
+  // TODO: based on usage, `fetched` and `synced` should be two different groups
```

Comment:
Any ticket to reference here, or is this something we're planning to do in the follow up settings pr?

Reply:
I'll open a ticket to not loose track (also yeah, I was planning to address it with follow-up changes, didn't want to introduce too much only partially related changes to the store in this PR)

Reply:
Okay, instead of creating a new one, I'll just added this ticket as a todo as I'm planning to just do this right away, hope that's fine from your perspective https://github.com/mongodb-js/compass/pull/4733/commits/0382345907add4c9ee6207a82dee35019be389aa

Reply:
Cool, was checking it wasn't a leftover todo that wasn't planned.